src/sdk/analytics/platform/vtex/index.ts

```diff
@@ -0,0 +1,47 @@
+import type { AnalyticsEvent } from '@faststore/sdk'
+
+import { send } from './search'
+
+const ONE_MINUTE_MS = 60 * 1e3
+
+setInterval(() => send({ type: 'session.ping' }), ONE_MINUTE_MS)
+
+const isFullTextSearch = (url: URL) =>
+  typeof url.searchParams.get('q') === 'string' &&
+  /^\/s(\/)?$/g.test(url.pathname)
+
+export default function sendEvent(event: AnalyticsEvent) {
+  // VTEX RC
+  window?.sendrc?.(event.name, event.params)
+
+  // VTEX Intelligent Search
+  if (event.name === 'search') {
+    send({
+      type: 'search.query',
+      text: event.params.search_term,
+      match: event.params.records_filtered,
+      url: event.params.url,
+    })
+  } else if (event.name === 'select_item') {
+    const url = new URL(event.params.url)
+
+    if (!isFullTextSearch(url)) {
+      return
+    }
+
+    for (const item of event.params.items ?? []) {
+      const productId = item.item_id ?? item.item_variant
+      const position = item.index
+
+      if (productId && position) {
+        send({
+          type: 'search.click',
+          productId,
+          position,
+          url: url.href,
+          text: url.searchParams.get('q') ?? '<empty>',
+        })
+      }
+    }
+  }
```

Comment:
To make the sendEvent function a little bit clearer, can you move this code to a new function?